pkg/controller/clone/planner.go

```diff
@@ -0,0 +1,493 @@
+package clone
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+
+	"github.com/go-logr/logr"
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	cc "kubevirt.io/containerized-data-importer/pkg/controller/common"
+)
+
+// Planner plans clone operations
+type Planner struct {
```

Comment:
So, do we plan to eventually reuse the planning/phase structure in the old clone flow? I assumed a lot of this new logic would be reused when we discussed this PR yesterday.

Reply:
Don't have the details nailed down yet, but it would be nice if datavolume controller can call `Plan`.  Strategy will always be host assisted in that case for pvc or snapshot source.  The main difference is that datavolume controller will not get a rebind phase